### PR TITLE
refactor: cleanup gradle process

### PR DESCRIPTION
**Pull request by erisu**
* Refactored the Gradle scripts by removing excessive logic.
* Updated the create script to create the Gradle `config.json` file out of the `defaults.json`.
* Updated the prepare script to 3-way-merge the `config.json` with `defaults.json` and user defined preferences from `config.xml`

**Exisiting `config.xml` Preferences:**

* android-minSdkVersion
* android-maxSdkVersion
* android-targetSdkVersion
* GradlePluginKotlinVersion
* AndroidXAppCompatVersion
* GradlePluginGoogleServicesVersion
* GradlePluginGoogleServicesEnabled
* GradlePluginKotlinEnabled

**New `config.xml` Preferences:**

* android-buildToolsVersion
* GradleVersion
* AndroidGradlePluginVersion

### CODE REVIEW

File: framework/cordova.gradle [hunk 1]
```diff
@@ -168,11 +152,17 @@ def doGetConfigPreference(name, defaultValue) {
 
 // Properties exported here are visible to all plugins.
 ext {
+    def jsonFile = new File("$rootDir/config.json")
```
Comment (erisu) [inline framework/cordova.gradle:155]: This needs to be confirmed.

Previously it was:
```
def jsonFile = new File("$rootDir/${getProperties().get('cordova.defaultsFileLocation')}/defaults.json")
```

I have tested:

* `npm t`
* Creating a project from CLI
* Building project from CLI

I have not confirmed with:
* platform-centered workflow

I believe this workflow was going to be removed in this release but raphinesse will need to confirm this as he has the PR in draft.

File: framework/project.properties [hunk 2]
```diff
@@ -5,8 +5,8 @@
 split.density=false
 
 # Project target.
-target=android-{{DEFAULT_SDK_VERSION}}
+target=android-{{SDK_VERSION}}
 apk-configurations=
 renderscript.opt.level=O0
 android.library=true
-cordova.defaultsFileLocation={{DEFAULTS_FILE_PATH}}
+cordova.defaultsFileLocation={{FILE_PATH}}
```
Comment (erisu) [inline framework/project.properties:12]: If the above changes are OK, `defaultsFileLocation` can be removed.

File: bin/lib/create.js [hunk 3]
```diff
@@ -67,18 +67,14 @@ function copyJsAndLibrary (projectPath, shared, projectName, targetAPI) {
...
-            DEFAULT_SDK_VERSION: targetAPI || constants.DEFAULT_SDK_VERSION,
-            DEFAULTS_FILE_PATH: './defaults.json'
-        });
-        TemplateFile.render(path.join(ROOT, 'framework', 'build.gradle'), path.join(nestedCordovaLibPath, 'build.gradle'), {
-            DEFAULT_MIN_SDK_VERSION: constants.DEFAULT_MIN_SDK_VERSION
-        });
-        TemplateFile.render(path.join(ROOT, 'framework', 'cordova.gradle'), path.join(nestedCordovaLibPath, 'cordova.gradle'), {
-            DEFAULT_BUILD_TOOLS_VERSION: constants.DEFAULT_BUILD_TOOLS_VERSION
+            SDK_VERSION: targetAPI || constants.SDK_VERSION,
+            FILE_PATH: './config.json'
```
Comment (erisu) [inline bin/lib/create.js:71]: If the changes are confirmed, this `FILE_PATH` can be removed.

File: framework/cordova.gradle (hunk 1 above)
Comment (breautek) [reply framework/cordova.gradle:155]: I never tested platform-centered workflow either with my original PR draft.

**Review by breautek — APPROVED** [review]

**Pull request merged**